encryptor.go

```diff
@@ -82,19 +86,32 @@ func WithCipher(cipher string) Option {
 	})
 }
 
+// WithCost sets the cipher key cost for the encryptor to 2^power overriding
+// the default value of 18 (ie. 2^18=262144). Higher values increases the
+// cost of an exhaustive search but makes encoding and decoding proportionally slower.
+// This should only be in testing as it affects security.
+func WithCost(_ *testing.T, costPower uint) Option {
```

Comment:
I think we should check to confirm that t is not nil, and panic if it is nil.  Although I'm not a fan of panicking in general, this is one of those situations where it makes sense.

Reply:
Cool, updated